apps/happy-blocks/src/pricing-plans/components/billing-options.tsx

```diff
@@ -1,34 +1,76 @@
-import { createInterpolateElement } from '@wordpress/element';
+import { TERM_ANNUALLY, TERM_MONTHLY } from '@automattic/calypso-products';
+import { sprintf, __ } from '@wordpress/i18n';
 import { FunctionComponent } from 'react';
-import { PricingPlanBilling } from '../types';
+import { BlockPlan } from '../hooks/pricing-plans';
 
 const Promo: FunctionComponent = ( { children } ) => (
-	<span className="hb-pricing-plans-embed__promo">({ children })</span>
+	<span className="hb-pricing-plans-embed__promo">{ children }</span>
 );
 
 interface Props {
-	billings: PricingPlanBilling[];
+	plans: BlockPlan[];
 	value: string;
-	onChange: ( value: string ) => void;
+	onChange: ( value?: string ) => void;
 }
 
-const BillingOptions: FunctionComponent< Props > = ( { billings, value, onChange } ) => {
+const getAnnyallyDiscount = ( annuallyPlan?: BlockPlan, monthlyPlan?: BlockPlan ) => {
+	if ( ! annuallyPlan || ! monthlyPlan ) {
+		return null;
+	}
+
+	const annualPricePerMonth = annuallyPlan.rawPrice / 12;
+
+	const discountRate = Math.round(
+		( 100 * ( monthlyPlan.rawPrice - annualPricePerMonth ) ) / monthlyPlan.rawPrice
+	);
+
+	return sprintf(
+		// translators: %s is the discount rate
+		__( 'Save %s', 'happy-blocks' ),
+		`${ discountRate }%`
+	);
+};
+
+const BillingOptions: FunctionComponent< Props > = ( { plans, value, onChange } ) => {
+	const selectedPlan = plans?.find( ( plan ) => plan.productSlug === value );
+
+	if ( ! plans || ! selectedPlan ) {
+		return null;
+	}
+
+	const monthlyPlan =
+		selectedPlan.term === TERM_MONTHLY
+			? selectedPlan
+			: plans.find( ( plan ) => plan.term === TERM_MONTHLY && plan.type === selectedPlan.type );
+
+	const annuallyPlan =
+		selectedPlan.term === TERM_ANNUALLY
+			? selectedPlan
+			: plans.find( ( plan ) => plan.term === TERM_ANNUALLY && plan.type === selectedPlan.type );
+
+	const annuallyDiscount = getAnnyallyDiscount( annuallyPlan, monthlyPlan );
+
+	const billings = [ monthlyPlan, annuallyPlan ].filter( Boolean );
```

Comment:
I know I haven't pointed this out before but to be honest, I don't think the word "billings" exists. 😅 Do you think it's worth refactoring it to just `billing` or using something else? Perhaps "plans" here would suffice?

Reply:
Sounds good, thanks.